packages/components/pie-assistive-text/src/index.ts

```diff
@@ -46,12 +35,12 @@ export class PieAssistiveText extends LitElement implements AssistiveTextProps {
             class="c-assistiveText"
             data-test-id="pie-assistive-text"
             variant=${ifDefined(variant)}>
-            ${this.renderIcon()}
-            <slot></slot>
-        </p>`;
+                ${variant === 'success' ? html`<icon-check-circle class="c-assistiveText-icon" size="s"></icon-check-circle>` : nothing}
+                ${variant === 'error' ? html`<icon-alert-circle class="c-assistiveText-icon" size="s"></icon-alert-circle>` : nothing}
```

Comment:
Extracting this to a function shouldn't on its own cause the rendering error (this is how we render the modal too and is a common pattern for breaking down large rendering logic). Is your observation that simply moving this conditional logic to a function causes the issue? If so this might be worth flagging with Lit